electrumsv/restapi.py

```diff
@@ -0,0 +1,262 @@
+import asyncio
+import json
+from base64 import b64decode
+from json import JSONDecodeError
+from typing import Optional, ClassVar, Dict, Union, Any, Tuple
+from aiohttp import web
+from aiohttp.web_urldispatcher import UrlDispatcher
+import logging
+from .constants import MAX_MESSAGE_BYTES
+from .app_state import app_state
+from .util import to_bytes, to_string, constant_time_compare
+
+
+def get_app_state():
+    # to monkeypatch app_state in tests
+    return app_state
+
+
+def get_network_type():
+    app_state = get_app_state()
+    if app_state.config.get('testnet'):
+        return 'test'
+    if app_state.config.get('scalingtestnet'):
+        return 'stn'
+    else:
+        return 'main'
+
+
+class Errors:
+    # http 400 bad requests
+    GENERIC_BAD_REQUEST_CODE = 40000
+    URL_INVALID_NETWORK_CODE = 40001
+    URL_NETWORK_MISMATCH_CODE = 40002
+    JSON_DECODE_ERROR_CODE = 40003
+    LOAD_BEFORE_GET_CODE = 40004
+    EMPTY_REQUEST_BODY_CODE = 40005
+    NO_COINS_CODE = 40006
```

Comment:
These codes should be moved up to the higher level dapp code

Reply:
I think you mean just the ones applicable to the dapp and not the more generic ones referenced in restapi.py - otherwise I will need to remove `fault_to_http_response` (which I can remove if you like) but also the middlewares for basic auth and checking the url for correct network format... there's probably a way I could still add them from the dapp but I don't think that's what you mean?